### PR TITLE
Add config per label and retry mechanism

### PR DESCRIPTION
**Pull request by pavlovic-ivan**
This PR introduces a config file in yml format, that is picked up by the backend, which then looks for the config params by workflow job label. If the label exists, self hosted runner will be deployed for that job. If the label does not exist, runner will not be deployed. Also, new retry mechanism is added to retry stack.refresh and stack.destroy actions.

Before merging, please add an environment variable (not a secret) to your `protected` environment:
- [x] env var name: `CONFIG`
- [x] value:
```
ubuntu-20.04-arm64:
  machineType: t4g.xlarge
  machineImage: gross-pwsh-libicu-ghrunner-23090-arm64
  bootDiskSizeInGB: 500
  bootDiskType: gp2
```
- [x] feel free to remove current env vars: `JOB_FILTER`, `MACHINE_IMAGE`, `MACHINE_TYPE`, `BOOT_DISK_SIZE_IN_GB`, `BOOT_DISK_TYPE`

### CODE REVIEW

File: .github/workflows/aws_sam.yml [hunk 1]
```diff
@@ -9,8 +9,12 @@ jobs:
   test:
     name: Test
     runs-on: ubuntu-latest
+    environment: protected
     steps:
       - uses: actions/checkout@v3
+
+      - name: Transform YAML to JSON and echo to file
```
Comment (jgiannuzzi) [inline .github/workflows/aws_sam.yml:16]: This is not transforming to json!

**Review by jgiannuzzi — COMMENTED** [review]
It looks like this PR was created from a branch that didn't start from the main branch of this repo, which makes it confusing to review as it also show changes in the retry mechanism (which I thought we merged already?)

File: .github/workflows/aws_sam.yml (hunk 1 above)
Comment (pavlovic-ivan) [reply .github/workflows/aws_sam.yml]: Renamed

**Comment by pavlovic-ivan** [comment]
> It looks like this PR was created from a branch that didn't start from the main branch of this repo, which makes it confusing to review as it also show changes in the retry mechanism (which I thought we merged already?)

It is merged. And this was weird for me as well to see. But i've noticed that each time i create a PR, the number of files in the PR is bigger and bigger, with changes already made. Maybe i made a mistake somewhere, but i am wondering if the reason for this is because i didn't merge back parquetsharp-ghrunner-app remote into my main branch, after PR is merged. What do you think?

What i do is, i create a branch from parquetsharp-ghrunner-app/main. Then i merge my main branch with the latest changes to this branch. I push to `origin ps-main`, and i create a PR. Maybe after PR is merged, i should merge parquetsharp-ghrunner-app/main into origin/main before continuing.

**Comment by jgiannuzzi** [comment]
You should actually recreate the branch entirely every time a merge is done here

**Comment by pavlovic-ivan** [comment]
I did actually. I delete it from my local repo, and from remote repo. Sometimes i give it the same name, sometimes i don't. This time i didn't

**Review by jgiannuzzi — APPROVED** [review]

**Pull request merged**